backend/src/od.rs

```diff
@@ -41,25 +51,42 @@ impl DemandModel {
         let mut rng = WyRand::new_seed(42);
         let mut requests = Vec::new();
 
-        for (zone1, zone2, raw_count) in &self.desire_lines {
-            // To speed up the impact calculation, how many specific requests per (zone1, zone2)? If
-            // true, just do one, but weight it by count.
-            let (iterations, trip_count) = if fast_sample {
-                (1, *raw_count)
+        fn choose(slice: &[RoadID], rng: &mut WyRand) -> Option<RoadID> {
+            if slice.is_empty() {
+                return None;
+            }
+            let idx = rng.generate_range(0..slice.len());
+            Some(slice[idx])
+        }
+
+        // One sample will be made for this many trips when `fast_sample` is enabled.
+        // Increasing this will speed up fast_sample mode, but give less accurate results.
+        // Some OD pairs with less than this many trips might be skipped completely.
+        let trips_per_sampled_request = 10;
```

Comment:
This could be much slower in areas with loads of trips, right, like maybe Edinburgh? Before with `fast_sample`, 5000 trips from zone1 to zone2 would just turn into one route. Now it'll do 500. I think you're right that this is the better approach -- the results from just sampling 1 were really biased and odd anyway.

Reply:
I just read the commit message and comment more closely -- this new way is indeed more fair

Reply:
We can definitely crank this sample factor up if we want to speed things up further. In my own measurements, it's to a point where this part of the code is ~50% of the runtime, so you'll eventually see diminishing returns as the time spent building a router takes over.